src/rtracer.js

```diff
@@ -221,7 +221,7 @@ const hapiPlugin = ({
 
     if (echoHeader) {
       server.ext('onPreResponse', async (request, h) => {
-        request.response.header(headerName, id())
+        request.raw.res.setHeader(headerName, id())
```

Comment:
I'd prefer to use framework's APIs when possible. This code should be enough:

```js
      server.ext('onPreResponse', async (request, h) => {
        const res = request.response
        if (res && res.isBoom) {
          res.output.headers[headerName] = id()
        } else if (res) {
          res.header(headerName, id())
        }
        return h.continue
      })
```

WDYT?

Reply:
I ended up not being able to get branch coverage for if `request.response` didn't exist when I had something like that. Maybe it's better to not have that check for something that probably can't exist to get to 100% coverage. If I find a way it fails in the future, I'll then know how to set up the test.